spec/unit/ec2_server_create_spec.rb

```diff
@@ -20,69 +20,133 @@
 require "net/ssh/proxy/http"
 require "net/ssh/proxy/command"
 require "net/ssh/gateway"
-require "fog/aws"
 require "chef/util/path_helper"
+require "aws-sdk-ec2"
+Chef::Knife::Bootstrap.load_deps
 
 describe Chef::Knife::Ec2ServerCreate do
   let(:knife_ec2_create) { Chef::Knife::Ec2ServerCreate.new }
-  let(:ec2_connection) { double(Fog::Compute::AWS) }
-  let(:ec2_servers) { double() }
+  let(:ec2_connection)   { Aws::EC2::Client.new(stub_responses: true) }
+  let(:ebs)              { OpenStruct.new(volume_size: 30, iops: 123) }
+  let(:groups)           { [OpenStruct.new(name: "grp-646rswg")] }
+  let(:placement)        { OpenStruct.new(tenancy: "default", group_name: "some_placement_group") }
+  let(:state)            { OpenStruct.new(name: "running") }
+  let(:security_groups)  { [OpenStruct.new(group_id: "s-gr446f", group_name: "default-vpc")] }
+  let(:tags)             { [OpenStruct.new(key: "Name", value: "ec2-test")] }
+  let(:block_device_mappings) { OpenStruct.new(ebs: ebs) }
+  let(:network_interfaces)    { OpenStruct.new(subnet_id: "subnet-9d4a7b6") }
+  let(:instance1) do
+    OpenStruct.new(
+      architecture: "x86_64",
+      image_id: "ami-005bdb005fb00e791",
+      instance_id: "i-00fe186450a2e8e97",
+      instance_type: "t2.micro",
+      key_name: "ssh_key_name",
+      platform: "windows",
+      name: "image-test",
+      description: "test windows winrm image",
+      block_device_mappings: [block_device_mappings],
+      network_interfaces: [network_interfaces],
+      groups: groups,
+      placement: placement,
+      state: state,
+      security_groups: security_groups,
+      tags: tags
+    )
```

Comment:
So as a general pattern this concerns me because OpenStruct is pretty terrible and instance_double would be better.  So please never copy this a general way to mock objects.  But if all the aws gem is passing back is an OpenStruct then you have to do this...

Reply:
Okay sure @lamont-granquist thanks!